### DISCUSSION TITLE
Upgrade from v3.2 to 4.0-beta4 fails

### DISCUSSION THREAD

**Discussion by Eichhorn-personal** [forkineye/ESPixelStick#693] [Q&A] [answered]
fairly new to coding and upgrading. I have a Renard Plus Pixel controller running v3.2. I've been trying to upgrade the firmware to v4.0 so I can use this with xLights. Every beta I've tried fails, either using ESPSFlashTool.jar or building the EFU and trying OTA update. 

After doing some research, it seems that the firmware is too big for the board to handle. Someone suggested that I try WLED. That firmware failed as well, until it was suggested I try the 2mb firmware, which worked fine, but I'm having issues with the wifi.

Is there a smaller version of the firmware for ESPixelStick, or is there someone who has experience with the Renard Plus controllers? I've seen videos of people doing the OTA update without an issue, so I'm not sure what I'm doing wrong. I'm not sure what technical version information needed to help diagnose.

Thank you in advance

**Comment by MartinMueller2003** [comment]
Renard Plus controller is not on the supported platform list. Can you send me a link to the HW you are using?

**Reply by Eichhorn-personal** [reply]
https://www.renard-shop.com/home/37-es-wifi-pixel-controller.html

The controller runs fine on v3.2
![Screenshot 2023-11-26 143215](https://github.com/forkineye/ESPixelStick/assets/64677523/5bb3151a-bb54-43ad-b71c-d2e4720cd53e

**Comment by MartinMueller2003** [comment, marked as answer]
That device has only 1MB of flash. V4 needs 4MB of flash. It looks like you have an ESP01 based controller which V4 does not support.

**Reply by Eichhorn-personal** [reply]
That answers that. Thank you for the response
